datafusion/physical-expr/src/window/partition_evaluator.rs

```diff
@@ -127,13 +117,16 @@ pub trait PartitionEvaluator: Debug + Send {
         Ok(())
     }
 
-    /// Sets the internal state for window function
+    /// When the window frame has a fixed beginning (e.g UNBOUNDED
```

Comment:
While writing this explanation it occurred to me that the term `memoize`, while technically accurate is not the entire story here. I wonder if there is a better name like `adjust_window_state` 🤔

Reply:
I think `adjust_window_state` feels like a more general functionality. However, this function is a special case for `nth_value` when window frame boundary has `UNBOUNDED PRECEDING`. Hence I think `memoize` is better for this case.

Reply:
I will leave it named `memoize`